src/app/auctioneer/auctioneer.service.ts

```diff
@@ -57,28 +57,27 @@ export class AuctioneerService {
                 'Authorization': 'Bearer ' + this.authService.getToken()
             })
         }
-        
+
         return this.httpClient.get<any>(`${environment.BASE_URL}/auctions`, httpOptions)
             .subscribe(response => {
                 if(response){
-                    
+
```

Comment:
Instead of having the return type as any, make it AuctionItem[] and send each item in the result array to the constructor of AuctionItem. That is the proper way to deserialize.

Reply:
I updated the deserialization to be implemented like you described. Let me know what you think.